mongo/mongo.go

```diff
@@ -181,6 +181,10 @@ func (m *Mongo) RoundTrip(msg *Message) (*Message, error) {
 	// TODO add support for one-way messages (unacked writes)
 	wm, err := m.roundTrip(conn, msg.Wm)
 	if err != nil {
+		// see https://github.com/mongodb/mongo-go-driver/blob/v1.3.2/x/mongo/driver/operation.go#L368-L370
+		if ep, ok := server.(driver.ErrorProcessor); ok {
```

Comment:
I'd recommend a stricter check. The driver has this conditional check because the deployment types used by the SDAM monitoring routines and during connection handshakes don't implement `ErrorProcessor`. However, the proxy should always be using a server of type `*topology.Server`, which must implement `ErrorProcessor`, so I'd recommend either failing or at least logging an error if `ok` is false.

Reply:
Good suggestion, added a warning log.